### DISCUSSION TITLE
Trying to understand  `--set-datapack-growfactor`/`--set-treepack-growfactor`

### DISCUSSION THREAD

**Discussion by kapitainsky** [rustic-rs/rustic#979] [Q&A] [answered]
The last question this year. I have been staring at  `--set-datapack-growfactor`/`--set-treepack-growfactor` explanation and I am not convinced I get it right.

Could somebody show be some sample calculation?

Let's say:

`--set-datapack-size <SIZE>` is default `32 MiB`
`--set-datapack-growfactor` is default 32

What will be pack size for 5TiB (5120GiB) repo?

SQRT(5120) = 71.55

So am I correct that pack size  in this case = `32 kiB * 32 * 71.55 = 73267 kiB = 72 MiB` ?

As per `rustic --help`:

```
The default packsize grows by the square root of the total size of all data packs multiplied with this factor. 
This means 32 kiB times this factor per square root of total datasize in GiB.
```

**Comment by simonsan** [comment]
As you seem to want to dive into the code as well, here is the calculation: https://github.com/rustic-rs/rustic_core/blob/853995e115da81535df4d5f79592990bf19b9df1/src/blob/packer.rs#L90

**Reply by kapitainsky** [reply]
Thx. 

I am still not 100% sure....

Would you mind to give me an example for 5TiB repo?

**Comment by aawsome** [comment, marked as answer]
> What will be pack size for 5TiB (5120GiB) repo?
> 
> SQRT(5120) = 71.55
> 
> So am I correct that pack size in this case = `32 kiB * 32 * 71.55 = 73267 kiB = 72 MiB` ?

Correct - your default datapack size is increased by 72MiB in this case. With a default size of 32 MiB this will result in targeted pack sizes of 104 MiB, then.

N.B: 32kiB * 32 = 1 MiB. This means the default is to increase the pack size by 1 MiB per `sqrt(total size in GiB)`. 

Some Examples using the default values:
- 1 GiB total size => 33 MiB targeted packsize
- 100 GiB total size => 42 MiB targeted packsize
- 1600 GiB total size => 72 MiB targetd packsize
